### ISSUE TITLE
test seems to ignore allowVersionFallback

### ISSUE THREAD

**Issue by morgantic** [meza/minecraft-mod-manager#491]
### What happened?

I ran mmm test and got the info
`Some mods are missing support for 1.20.2.
❌ AdvancementInfo (G1epq3jN)
❌ Inventory Sorting (5ibSyLAz)
You cannot upgrade to 1.20.2 just yet.`

### What did you expect to happen?

Both mods are available for 1.20.1, and since I have `"allowVersionFallback": true` set in the config, I expected this to not hinder the upgrade process

### Which command did you run?

mmm test

### Operating System

Windows

### What version of MMM are you running?

error: unknown command 'version' (the currently latest, tho)

### Contents of the modlist.json

```json
{
  "loader": "fabric",
  "gameVersion": "1.20.1",
  "allowVersionFallback ": true,
  "defaultAllowedReleaseTypes": [
    "beta",
    "release"
  ],
  "modsFolder": "./mods",
  "mods": [
    {
      "type": "modrinth",
      "id": "G1epq3jN",
      "name": "AdvancementInfo"
    },
    {
      "type": "curseforge",
      "id": "495453",
      "name": "Axolotl Bucket Fix"
    },
    {
      "type": "modrinth",
      "id": "rcLriA4v",
      "name": "Amecs"
    },
    {
      "type": "modrinth",
      "id": "kYq5qkSL",
      "name": "Borderless Mining"
    },
    {
      "type": "modrinth",
      "id": "VX3TgwQh",
      "name": "Carpet Extra"
    },
    {
      "type": "modrinth",
      "id": "Orvt0mRa",
      "name": "Indium"
    },
    {
      "type": "modrinth",
      "id": "5ibSyLAz",
      "name": "Inventory Sorting"
    },
    {
      "type": "modrinth",
      "id": "M0aimenU",
      "name": "Isometric Renders"
    },
    {
      "type": "modrinth",
      "id": "fQEb0iXm",
      "name": "Krypton"
    },
    {
      "type": "curseforge",
      "id": "244260",
      "name": "MiniHUD"
    },
    {
      "type": "modrinth",
      "id": "ko8Qabo1",
      "name": "Notes"
    },
    {
      "type": "modrinth",
      "id": "PtjYWJkn",
      "name": "Sodium Extra"
    },
    {
      "type": "modrinth",
      "id": "H8CaAYZC",
      "name": "Starlight (Fabric)"
    },
    {
      "type": "modrinth",
      "id": "b93awgkg",
      "name": "Status"
    },
    {
      "type": "modrinth",
      "id": "IOYm9Bdm",
      "name": "WAIG: Where Am I Going"
    },
    {
      "type": "curseforge",
      "id": "297344",
      "name": "Tweakeroo"
    },
    {
      "type": "modrinth",
      "id": "lhGA9TYQ",
      "name": "Architectury API"
    },
    {
      "type": "modrinth",
      "id": "yBW8D80W",
      "name": "LambDynamicLights"
    },
    {
      "type": "curseforge",
      "id": "303119",
      "name": "MaLiLib"
    },
    {
      "type": "modrinth",
      "id": "gvQqBUqZ",
      "name": "Lithium"
    },
    {
      "type": "curseforge",
      "id": "308892",
      "name": "Litematica"
    },
    {
      "type": "modrinth",
      "id": "ccKDOlHs",
      "name": "oωo (owo-lib)"
    },
    {
      "type": "modrinth",
      "id": "EsAfCjCV",
      "name": "AppleSkin"
    },
    {
      "type": "modrinth",
      "id": "NcUtCpym",
      "name": "Xaero's World Map"
    },
    {
      "type": "modrinth",
      "id": "1IjD5062",
      "name": "Continuity"
    },
    {
      "type": "modrinth",
      "id": "8shC1gFX",
      "name": "BetterF3"
    },
    {
      "type": "modrinth",
      "id": "P7dR8mSH",
      "name": "Fabric API"
    },
    {
      "type": "modrinth",
      "id": "P7dR8mSH",
      "name": "Fabric API"
    },
    {
      "type": "modrinth",
      "id": "nfn13YXA",
      "name": "Roughly Enough Items (REI)"
    },
    {
      "type": "modrinth",
      "id": "FOIvwGKz",
      "name": "Sounds Be Gone!"
    },
    {
      "type": "modrinth",
      "id": "n6PXGAoM",
      "name": "Better Statistics Screen"
    },
    {
      "type": "modrinth",
      "id": "l6YH9Als",
      "name": "spark"
    },
    {
      "type": "curseforge",
      "id": "225608",
      "name": "WorldEdit"
    },
    {
      "type": "modrinth",
      "id": "9eGKb6K1",
      "name": "Simple Voice Chat"
    },
    {
      "type": "modrinth",
      "id": "mOgUt4GM",
      "name": "Mod Menu"
    },
    {
      "type": "modrinth",
      "id": "9s6osm5g",
      "name": "Cloth Config API"
    },
    {
      "type": "modrinth",
      "id": "TQTTVgYE",
      "name": "Carpet"
    },
    {
      "type": "modrinth",
      "id": "YL57xq9U",
      "name": "Iris Shaders"
    },
    {
      "type": "modrinth",
      "id": "AANobbMI",
      "name": "Sodium"
    }
  ]
}
```

### I have updated

- [X] I swear I have tried it with the latest version!

**Comment by meza** [comment]
This release _should_ fix this: https://github.com/meza/minecraft-mod-manager/releases/tag/v1.3.7

Can you verify? I didn't explicitly try to fix this but what I was working on should™️ do the trick

**Comment by morgantic** [comment]
I think I found the issue. The config file reads "allowVersionFallback ": true, not "allowVersionFallback": true - note the surplus whitespace. I don't know how it got in there, but removing it seems to fix the issue (I can't reproduce it exactly because I don't have the 1.20.1 modlist anymore)

**Comment by morgantic** [comment]
Ok, so while the surplus whitespace was very much an issue, I also copied the wrong mmm executable over, which fortunately led me to perfectly test the 1.2.7 bugfix!

So this is with 1.3.6:
`MzS3-1_20-scan> .\mmm.exe test
Some mods are missing support for 1.20.4.
❌ Axolotl Bucket Fix (495453)
❌ WorldEdit (225608)
❌ Amecs (rcLriA4v)
❌ Borderless Mining (kYq5qkSL)
❌ Carpet Extra (VX3TgwQh)
❌ Indium (Orvt0mRa)
❌ Isometric Renders (M0aimenU)
❌ Krypton (fQEb0iXm)
❌ Notes (ko8Qabo1)
❌ WAIG: Where Am I Going (IOYm9Bdm)
❌ LambDynamicLights (yBW8D80W)
❌ Lithium (gvQqBUqZ)
❌ oωo (owo-lib) (ccKDOlHs)
❌ AppleSkin (EsAfCjCV)
❌ Continuity (1IjD5062)
❌ BetterF3 (8shC1gFX)
❌ Sounds Be Gone! (FOIvwGKz)
❌ Better Statistics Screen (n6PXGAoM)
You cannot upgrade to 1.20.4 just yet.`

Then I fixed the surplus whitespace (still 1.3.6)
`MzS3-1_20-scan> .\mmm.exe test
Some mods are missing support for 1.20.4.
❌ Notes (ko8Qabo1)
You cannot upgrade to 1.20.4 just yet.`

Then I realized the version mistake and upgraded:
`MzS3-1_20-scan> .\mmm.exe -V
1.3.6
MzS3-1_20-scan> .\mmm.exe -V
1.3.7
MzS3-1_20`

Now:
`MzS3-1_20-scan> .\mmm.exe test
All mods have support for 1.20.4. You can safely upgrade.`

So, can confirm that the 1.3.7 bugfix works!
